src/lib.rs

```diff
@@ -288,32 +347,37 @@ impl DevServer {
         Ok(())
     }
 
-    pub fn serve_and_watch(
-        self,
-        build_dir_path: impl AsRef<Path>,
-        command: process::Command,
-    ) -> Result<()> {
-        let build_dir_pathbuf = build_dir_path.as_ref().to_owned();
-        let mut watch = self.watch.clone();
+    pub fn start(self, watch: Option<Watch>) -> Result<()> {
```

Comment:
Move this back to a field. It's the command that needs to be optional, not the watch.

Here you lost all the watch CLI parameters in the dev server command. This is not good